### PR TITLE
Vtapfinal

### PR DESCRIPTION
**Pull request by karthikananth**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/virtualNetworkTap.json [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+                },
+                {
+                    "name": "tapName",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The name of the virtual network tap."
+                },
+                {
+                    "$ref": "./network.json/parameters/ApiVersionParameter"
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/virtualNetworkTap.json]: Should be "$ref": "./network.json#/parameters/ApiVersionParameter"

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/virtualNetworkTap.json [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+                },
+                {
+                    "$ref": "./network.json/parameters/SubscriptionIdParameter"
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/virtualNetworkTap.json]: Same here

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/virtualNetworkTap.json (hunk 1 above)
Comment (karthikananth) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/virtualNetworkTap.json]: resolved

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/virtualNetworkTap.json (hunk 2 above)
Comment (karthikananth) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/virtualNetworkTap.json]: resolved

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/virtualNetworkTap.json [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+          "networkInterfaceTapConfigurations": {
+            "type": "array",
+            "items": {
+                "$ref": "./networkInterface.json#/definitions/NetworkInterfaceTapConfiguration",
+                "description": "The reference of the Network Interface."
+            },
+            "description": "Specifies the list of resource IDs for the network interface IP configuration that needs to be tapped."
+          },
+          "destinationNetworkInterfaceIPConfiguration": {
+            "$ref": "./networkInterface.json#/definitions/IPConfiguration",
```
Comment (jianghaolu) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/virtualNetworkTap.json]: @anuchandy this is the ip configuration PUT on vtap requires

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkInterface.json [hunk 4]
```diff
@@ -0,0 +1,1352 @@
...
+          "description": "The reference of the public IP resource."
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "Gets the provisioning state of the public IP resource. Possible values are: 'Updating', 'Deleting', and 'Failed'."
+        }
+      },
+      "description": "Properties of IP configuration."
+    },
+    "IPConfiguration": {
```
[the remark this answers is not in the document]
Comment (karthikananth) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkInterface.json]: resolved

**Comment by sergey-shandar** [comment]
@karthikananth please, resolve conflicts.

**Comment by lmazuel** [comment]
@AutorestCI rebuild azure-sdk-for-python

**Comment by lmazuel** [comment]
@AutorestCI rebuild azure-sdk-for-python

**Comment by lmazuel** [comment]
@AutorestCI rebuild azure-sdk-for-python

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/NetworkInterfaceTapConfigurationCreate.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    "api-version": "2018-08-01",
+    "parameters": {
+      "properties": {
+        "virtualNetworkTap": {
+            "id": "/subscriptions/subid/resourceGroups/testrg/providers/Microsoft.Network/virtualNetworkTaps/testvtap"
+          }
+        },
+        "location": "centraluseuap" 
+    },
+    "responses" : {
```
Comment (sergey-shandar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/NetworkInterfaceTapConfigurationCreate.json]: These are not valid examples. `"responses"` should be outside of parameters.

**Review by sergey-shandar — CHANGES_REQUESTED** [review]

**Review by anuchandy — APPROVED** [review]

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**